test/extensions/filters/network/zookeeper_proxy/zookeeper_filter_test.cc

```diff
@@ -0,0 +1,709 @@
+#include "common/buffer/buffer_impl.h"
+
+#include "extensions/filters/network/zookeeper_proxy/zookeeper_decoder.h"
+#include "extensions/filters/network/zookeeper_proxy/zookeeper_filter.h"
+
+#include "test/mocks/network/mocks.h"
+#include "test/test_common/test_base.h"
+
+#include "gmock/gmock.h"
+
+using testing::NiceMock;
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace ZooKeeperProxy {
+
+bool protoMapEq(const ProtobufWkt::Struct& obj, const std::map<std::string, std::string>& rhs) {
+  EXPECT_TRUE(rhs.size() > 0);
+  for (auto const& entry : rhs) {
+    EXPECT_EQ(obj.fields().at(entry.first).string_value(), entry.second);
+  }
+  return true;
+}
+
+MATCHER_P(MapEq, rhs, "") { return protoMapEq(arg, rhs); }
+
+class ZooKeeperFilterTest : public TestBase {
+public:
+  ZooKeeperFilterTest() { ENVOY_LOG_MISC(info, "test"); }
+
+  void initialize() {
+    config_ = std::make_shared<ZooKeeperFilterConfig>(stat_prefix_, scope_);
+    filter_ = std::make_unique<ZooKeeperFilter>(config_);
+    filter_->initializeReadFilterCallbacks(filter_callbacks_);
+  }
+
+  Buffer::OwnedImpl encodeConnect(const bool readonly = false, const uint64_t zxid = 100,
+                                  const uint32_t session_timeout = 10,
+                                  const uint32_t session_id = 200,
+                                  const std::string& passwd = "") const {
```

Comment:
This is just test so it doesn't really matter, but pretty sure this will allocate a new string every time you call `encodeConnect` without specifying `passwd`